% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}

\usepackage{color}
\usepackage{framed}
\setlength{\fboxsep}{.8em}

% These colours were manually entered, they shouldn't matter unless you want pdf output

\newenvironment{redbox}{
  \definecolor{shadecolor}{RGB}{243, 154, 157}
  \color{white}
  \begin{shaded}}
 {\end{shaded}}

\newenvironment{bluebox}{
  \definecolor{shadecolor}{RGB}{172, 210, 237}
  \color{white}
  \begin{shaded}}
 {\end{shaded}}

\newenvironment{greenbox}{
  \definecolor{shadecolor}{RGB}{141, 181, 128}
  \color{white}
  \begin{shaded}}
 {\end{shaded}}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[]{natbib}
\bibliographystyle{plainnat}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Machine Learning 2023},
  pdfauthor={Faculty: David Wishart, Vasu Gautam, Mark Berjanskii, Sagan Girod, Michelle Brazas, Nia Hughes},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{Machine Learning 2023}
\author{Faculty: David Wishart, Vasu Gautam, Mark Berjanskii, Sagan Girod, Michelle Brazas, Nia Hughes}
\date{August 16, 2023 - August 17, 2023}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\part{Introduction}\label{part-introduction}

\chapter{Workshop Info}\label{workshop-info}

Welcome to the 2023 Machine Learning Canadian Bioinformatics Workshop webpage!

\section{Class Photo}\label{class-photo}

\section{Schedule}\label{schedule}

\begin{longtable}[]{@{}
  >{\centering\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.1064}}
  >{\centering\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.3936}}
  >{\centering\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.1064}}
  >{\centering\arraybackslash}p{(\columnwidth - 6\tabcolsep) * \real{0.3936}}@{}}
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\centering
\textbf{Time (Eastern)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\centering
\textbf{Wednesday, August 16}
\end{minipage} & \begin{minipage}[b]{\linewidth}\centering
\textbf{Time (Eastern)}
\end{minipage} & \begin{minipage}[b]{\linewidth}\centering
\textbf{Thursday, August 17}
\end{minipage} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
9:45 & Virtual Arrivals & 9:45 & Virtual Arrivals \\
10:00 & Welcome and Technology Check (Nia Hughes) & 10:00 & Module 5: Gene Prediction with NNs (Lecture and Lab) \\
10:45 & Module 1: Introduction to Machine Learning (Lecture) & 12:00 & Break (30min) \\
12:15 & Break (30min) & 12:30 & Module 6: Machine Learning with Keras and Scikit-learn (Lecture and Lab) \\
12:45 & Module 2: Decision Trees (Lecture and Lab) & 14:00 & Break (1hr) \\
14:15 & Break (45min) & 15:00 & Module 7 (Continued): Machine Learning with Keras and Scikit-Learn \\
15:00 & Module 3: Neural Networks (Lecture and Lab) & 16:00 & Break (30min) \\
16:30 & Break (30min) & 16:30 & Module 8: Information Extraction with ChatGPT (Lecture and Lab) \\
17:00 & Module 4: Neural Networks for Secondary Structure (Lecture and Homework) & 17:45 & Survey and Closing Remarks \\
18:00 & End of Day 1 & 18:00 & End of Day 2 \\
\end{longtable}

\section{Pre-work}\label{pre-work}

\href{https://docs.google.com/forms/d/e/1FAIpQLSckky4be53s62TkKLVMiTeOr3Rw0lwA5xN1rBkyExM3qEZIVA/viewform}{You can find your pre-work here.}

\chapter{Meet Your Faculty}\label{meet-your-faculty}

\subsubsection{David Wishart}\label{david-wishart}

\begin{quote}
Distinguished University of Biological Sciences and Computing Science
University of Alberta
Edmonton, AB, CA

--- \href{mailto:dwishart@ualberta.ca}{\nolinkurl{dwishart@ualberta.ca}}
\url{www.wishartlab.com}
\end{quote}

David was one of the co-founders of the CBW in 1998. He has active
research interests in the application of machine learning to a wide range of computational
biology problems, from genomics to proteomics to metabolomics. In addition to running a large
computational biology lab, David also operates a large wet lab (analytical chemistry, molecular
biology, nanotechnology, cell biology, structural biology) supported by its own fabrication shop
and electrical engineering group.

\subsubsection{Vasu Gautam}\label{vasu-gautam}

\begin{quote}
Senior Scientist
Wishart Lab, University of Alberta
Edmonton, AB, CA

--- \href{mailto:vasuk@ualberta.ca}{\nolinkurl{vasuk@ualberta.ca}}
\end{quote}

Dr.~Vasu Gautam is the senior scientist and Bioinformatics Manager at Wishart Node, University
of Alberta. Vasu is intrigued by the diverse world of ``omics'' and their combined role in biological
research, be it proteomics, genomics, or metabolomics. Vasu has worked in both academia and
industry in the field of multi-omics. His interest has been to explore the different aspects of these
``omics'' groups and then combine this knowledge pool to address some of the most difficult
questions in the field. Bioinformatics/computational biology has been a great tool in enhancing
this capability and continuing his research. His current focus is the study of machine learning
algorithms and their applications in different areas of metabolomics.

\subsubsection{Mark Berjanskii}\label{mark-berjanskii}

\begin{quote}
Research Associate
Department of the Biological Sciences
University of Alberta
Edmonton, AB, CA

--- \href{mailto:mb1@ualberta.ca}{\nolinkurl{mb1@ualberta.ca}}
\end{quote}

Mark obtained a Ph.D.~in Biochemistry at University of Missouri-Columbia, USA. Mark's
research interests include NMR metabolomics, bioinformatics, protein NMR structure
determination, interactions, misfolding, and dynamics. Between 1996 and 2004, he worked as a
member of several research teams that studied the Hsp70-Hsp40 chaperone system. Mark
joined Dr.~Wishart's group at University of Alberta in 2004. Between 2005 and 2013, Mark
studied prion proteins that, when misfolded, cause Mad Cow Disease in cattle and similar
Creutzfeldt-Jakob Disease in humans. Since 2004, he has been involved in developing several
programs for analysis of protein structure and dynamics, such as Random Coil Index, Preditor,
GeNMR, CS23D, PROSESS, Resolution-by-proxy, and Gamdy, as well as bioinformatic
analysis and NMR metabolomics.

\subsubsection{Sagan Girod}\label{sagan-girod}

\begin{quote}
Junior Data Scientist
Wishart Lab, University of Alberta
Edmonton, AB, CA

--- \href{mailto:lgirod@ualberta.ca}{\nolinkurl{lgirod@ualberta.ca}}
\end{quote}

Sagan is a data scientist at the Wishart Node, University of Alberta. His
research is based around machine learning and multi-omics with a focus on data curation and
database design.

\subsubsection{Michelle Brazas}\label{michelle-brazas}

\begin{quote}
Acting Scientific Director
Canadian Bioinformatics Workshops (CBW)
Toronto, ON, CA

--- \href{mailto:support@bioinformatics.ca}{\nolinkurl{support@bioinformatics.ca}}
\end{quote}

Dr.~Michelle Brazas is the Associate Director for Adaptive Oncology at the Ontario
Institute for Cancer Research (OICR), and acting Scientific Director at Bioinformatics.ca.
Previously, Dr.~Brazas was the Program Manager for Bioinformatics.ca and a faculty member in
Biotechnology at BCIT. Michelle co-founded and runs the Toronto Bioinformatics User Group
(TorBUG) now in its 11th season, and plays an active role in the International Society of
Computational Biology where she sits on the Board of Directors and Executive Board.

\subsubsection{Nia Hughes}\label{nia-hughes}

\begin{quote}
Program Manager, Bioinformatics.ca
Ontario Institute for Cancer Research
Toronto, ON, Canada

--- \href{mailto:nia.hughes@oicr.on.ca}{\nolinkurl{nia.hughes@oicr.on.ca}}
\end{quote}

Nia is the Program Manager for Bioinformatics.ca, where she coordinates
the Canadian Bioinformatics Workshop Series. Prior to starting at OICR, she completed her
M.Sc. in Bioinformatics from the University of Guelph in 2020 before working there as a
bioinformatician studying epigenetic and transcriptomic patterns across maize varieties.

\chapter{Data}\label{data}

\subsubsection{Course data downloads}\label{course-data-downloads}

Download the scripts and data for this course \href{https://drive.google.com/file/d/1MeMJ7B3Au8-a_w7ogSECZayb49Km961k/view?usp=sharing}{here}

\part{Modules}\label{part-modules}

\chapter{Module 1: Introduction to Machine Learning}\label{module-1-introduction-to-machine-learning}

\section{Lecture}\label{lecture}

\chapter{Module 2: Decision Trees}\label{module-2-decision-trees}

\section{Lecture}\label{lecture-1}

\chapter{Module 3: Neural Networks}\label{module-3-neural-networks}

\section{Lecture}\label{lecture-2}

\chapter{Module 4: Neural Networks for Secondary Structure}\label{module-4-neural-networks-for-secondary-structure}

\section{Lecture}\label{lecture-3}

\chapter{Module 5: Gene Finding with NNs}\label{module-5-gene-finding-with-nns}

\section{Lecture}\label{lecture-4}

\chapter{Module 6: Machine Learning with Keras and Scikit-Learn}\label{module-6-machine-learning-with-keras-and-scikit-learn}

\section{Lecture}\label{lecture-5}

\chapter{Module 7: Machine Learning with Keras and Scikit-Learn Continued}\label{module-7-machine-learning-with-keras-and-scikit-learn-continued}

\section{Lecture}\label{lecture-6}

\chapter{Module 8: Information Extraction with ChatGPT}\label{module-8-information-extraction-with-chatgpt}

\section{Lecture}\label{lecture-7}

  \bibliography{book.bib,packages.bib}

\end{document}
